### PR TITLE
Add flag to enable building with chakra core on uwp

### PR DESCRIPTION
**Pull request by stecrain**
Turns out there is already a mechanism to import react.overrides.props from the consuming app.
The idea is adding the property <CHAKRACOREUWP>true</CHAKRACOREUWP> in react.overrides.props to allow building with chakracore.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/2630)

### CODE REVIEW

**Comment by stecrain** [comment]
Just realized I forgot to include the changes to link chakracore.lib. adding those soon

**Comment by stecrain** [comment]
> You'll need to merge my latest checkin to get build verification. (sorry I broke the non-fork RN slice earlier)

No worries. TY for the quick review!

File: vnext/Chakra/ChakraHelpers.cpp [hunk 1]
```diff
@@ -49,6 +49,7 @@ bool fwrite(const T& val, FILE* file) noexcept
   return fwrite(&val, 1, file) == 1;
 }
 
+#if !defined(CHAKRACOREUWP)
```
Comment (akrantz) [inline vnext/Chakra/ChakraHelpers.cpp]: I would like to suggest making the define `CHAKRACORE_UWP`.

Comment (stecrain) [reply vnext/Chakra/ChakraHelpers.cpp]: whoops. did catch this comment before the automerge. I'll send a follow-up renaming this

**Pull request merged**